capstone-project/src/main/angular-webapp/src/app/map/map.component.spec.ts

```diff
@@ -22,4 +47,73 @@ describe('MapComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should center on user location', () => {
+    spyOn(navigator.geolocation, 'getCurrentPosition').and.callFake(function(success) {
+      const position = {
+        coords: {
+          accuracy: 1,
+          altitude: 1,
+          altitudeAccuracy: 1,
+          heading: 1,
+          speed: 0,
+          latitude: 20,
+          longitude: 85
+        }, timestamp: 1
+      };
+      success(position);
+    });
+
+    component.focusOnUserLocation();
+    const location = new google.maps.LatLng(20, 85)
+    
+    expect(component["gMap"].getCenter().lat()).toEqual(location.lat());
+    expect(component["gMap"].getCenter().lng()).toEqual(location.lng());
```

Comment:
expect(component["gMap"].getCenter()).toEqual(expectedLocation)

Reply:
I'm not sure why, but this fails. This is the error -
Expected $.lat = Function to equal Function.
Expected $.lng = Function to equal Function.
It could be because "setCenter" method changes fields, but I'm not sure. That is why I think it would be simpler to only check the coordinates.